manage_breast_screening/utils/date_formatting.py

```diff
@@ -0,0 +1,94 @@
+"""
+Format and dates and times in a way consistent with
+the NHS service manual's content guide:
+https://service-manual.nhs.uk/content/numbers-measurements-dates-time
+"""
+
+from datetime import date, datetime
+
+
+def format_date(value):
+    """
+    Format a date
+
+    >>> from datetime import date
+    >>> format_date(date(2025, 1, 1))
+    '1 January 2025'
+    """
+    return value.strftime("%-d %B %Y")
+
+
+def format_relative_date(value: datetime | date):
+    """
+    Format a date relative to today.
+    """
+    if isinstance(value, datetime):
+        value = value.date()
+
+    today = date.today()
+    if value == today:
+        return "today"
+
+    days = (value - today).days
+
+    if -6 <= days <= -2:
+        return f"{abs(days)} days ago"
+    elif days == -1:
+        return "yesterday"
+    elif days == 1:
+        return "tomorrow"
+    elif 2 <= days <= 6:
+        return f"in {days} days"
+    else:
+        return format_date(value)
+
+
+def format_date_time(value):
```

Comment:
Wonder if this should be making use of the `format_time` here for consistency. The prototype not doing it is _probably_ a bug.

Reply:
There is in an inconsistency in the format string - this one doesn't include the AM/PM, so I'll add that in. I'm not bothered about the implementation so much as the tests being correct.